hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/OMConfigKeys.java

```diff
@@ -357,6 +357,12 @@ private OMConfigKeys() {
       "ozone.om.s3.grpc.server_enabled";
   public static final boolean OZONE_OM_S3_GRPC_SERVER_ENABLED_DEFAULT =
       false;
+
+  public static final String OZONE_OM_NAMESPACE_S3_RULE_COMPLIANT =
+          "ozone.om.s3.naming.rule_compliant";
```

Comment:
I think this should mention "buckets" somewhere and "rule_compliant" could be improved.  Please avoid `_` if possible.

It also needs to be added to `ozone-default.xml`.

Reply:
>Please avoid _ if possible.
+1

Reply:
Later we'll have another PR that makes ozone volume to be able to contain non-s3 character. So I'm thinking if we should have two parameters to configure the strictS3 value or just one~  open to all ideas!

Reply:
We'll decide to use single parameter for both volume and bucket!